// @node-wot/binding-mqtt/dist/mqtt.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MqttForm = exports.MqttQoS = exports.MqttBrokerServer = exports.MqttsClientFactory = exports.MqttClientFactory = exports.MqttClient = void 0;
const td_tools_1 = require("@node-wot/td-tools");
var mqtt_client_1 = require("./mqtt-client");
Object.defineProperty(exports, "MqttClient", { enumerable: true, get: function () { return __importDefault(mqtt_client_1).default; } });
var mqtt_client_factory_1 = require("./mqtt-client-factory");
Object.defineProperty(exports, "MqttClientFactory", { enumerable: true, get: function () { return __importDefault(mqtt_client_factory_1).default; } });
var mqtts_client_factory_1 = require("./mqtts-client-factory");
Object.defineProperty(exports, "MqttsClientFactory", { enumerable: true, get: function () { return __importDefault(mqtts_client_factory_1).default; } });
var mqtt_broker_server_1 = require("./mqtt-broker-server");
Object.defineProperty(exports, "MqttBrokerServer", { enumerable: true, get: function () { return __importDefault(mqtt_broker_server_1).default; } });
__exportStar(require("./mqtt-client"), exports);
__exportStar(require("./mqtt-client-factory"), exports);
__exportStar(require("./mqtts-client-factory"), exports);
__exportStar(require("./mqtt-broker-server"), exports);
var MqttQoS;
(function (MqttQoS) {
    MqttQoS[MqttQoS["QoS0"] = 0] = "QoS0";
    MqttQoS[MqttQoS["QoS1"] = 1] = "QoS1";
    MqttQoS[MqttQoS["QoS2"] = 2] = "QoS2";
})(MqttQoS = exports.MqttQoS || (exports.MqttQoS = {}));
class MqttForm extends td_tools_1.Form {
    constructor() {
        super(...arguments);
        this["mqtt:qos"] = MqttQoS.QoS0;
    }
}
exports.MqttForm = MqttForm;
